### PR TITLE
Cothority TypeScript

### PR DESCRIPTION
**Pull request by Gilthoniel**
This pull request migrates the current cothority javascript library from the typescript version done for personhood.online. It cleans up the code, makes it more comparable with the Java library and adds support for testing, documentation and coverage.

### CODE REVIEW

File: external/js/cothority/spec/support/public.toml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+
```
Comment (jeffallen) [inline external/js/cothority/spec/support/public.toml]: Are these exactly the same as the file in external/docker/public.toml? I can see why it need to be repeated here, but it's a bummer. Wonder if we could copy it over via some magic in package.json's build/test rules? For later...

**Comment by jeffallen** [comment]
@kc1212 could you please review the part in proof.ts where proofs are checked? That seems like a critical part of the client, and I don't feel qualified to notice traps there.

File: external/js/cothority/spec/support/public.toml (hunk 1 above)
Comment (Gilthoniel) [reply external/js/cothority/spec/support/public.toml]: oh yeah thanks for the reminder, I need to do something about that

**Comment by kc1212** [comment]
`proof.ts` looks good to me, hopefully you didn't waste too much time on the use of index @Gilthoniel (it's `i+1` in Go and `i` in Java because Go uses something like "foreach" and java just uses the normal for-loop, so the final `i` is different. It's correctly implemented in the TS code coz it also uses the normal for-loop)

**Comment by Gilthoniel** [comment]
A few heads up for the R4M:
- Generated files in Kyber (dist, doc) have been removed and those will be generated on demand like before publishing. That allows much cleaner and easier PRs.
- Proof checking and block validation has been added
- Travis test has been enabled but PoP tests have been disabled until the personhood.online branch has been merged
- TSLint has been added and is running before the tests

**Pull request merged**